module/actor/sheet/actor-sheet.js

```diff
@@ -414,6 +486,13 @@ export default class PBActorSheet extends (foundry.appv1?.sheets?.ActorSheet ??
 
     formData.data.localizedType = `TYPES.${formData.actor.documentName}.${formData.data.type}`;
 
+    // Calculate effective speed for display
+    if (formData.data.type === 'character') {
+      const baseSpeed = formData.data.system.attributes?.speed?.max || 6;
```

Comment:
```suggestion
      const baseSpeed = formData.data.system.attributes?.speed?.max ?? 6;
```

Use the nullish version similar to the effective speed?